### PR TITLE
Hud refdrop

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Now hud_overlay and arranged_hud_process drops the reference on deletion.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!bugfix

File: code/defines/procs/hud.dm [hunk 1]
```diff
@@ -12,6 +12,25 @@ the HUD updates properly! */
...
+
+
+/image/hud_overlay/Destroy()
+
+	if(owner)
+		owner?.client?.images -= src
+		owner = null
+
+	. = ..()
+
```
Comment (SleepyGemmy) [inline code/defines/procs/hud.dm:33]:
```suggestion
	///Owner of the hud_overlay, aka who has the overlay
	var/mob/living/owner = null

/image/hud_overlay/New(icon, loc, icon_state, layer, dir)
	. = ..()

	if(ismob(loc))
		owner = loc

/image/hud_overlay/Destroy()
	if(owner)
		owner?.client?.images -= src
		owner = null

	. = ..()

```

File: code/defines/procs/hud.dm [hunk 2]
```diff
@@ -55,6 +74,13 @@ the HUD updates properly! */
 	var/mob/Mob
 	var/turf/Turf
 
+/datum/arranged_hud_process/Destroy(force)
+	Client = null
+	Mob = null
+	Turf = null
+	. = ..()
+
+
```
Comment (SleepyGemmy) [inline code/defines/procs/hud.dm:83]:
```suggestion
```

**Comment by SleepyGemmy** [comment]
!review

**Review by SleepyGemmy — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]

**Pull request merged**